sparkmagic/sparkmagic/livyclientlib/reliablehttpclient.py

```diff
@@ -24,6 +24,8 @@ def __init__(self, endpoint, headers, retry_policy):
             self._auth = HTTPKerberosAuth(mutual_authentication=REQUIRED)
         elif self._endpoint.auth_type == constants.AUTH_BASIC:
             self._auth = (self._endpoint.username, self._endpoint.password)
+        elif self._endpoint.auth_type != constants.NO_AUTH:
+            raise ValueError("Unsupported auth_type.")
```

Comment:
Please replace ValueError for Bad Config exception at https://github.com/jupyter-incubator/sparkmagic/blob/master/sparkmagic/sparkmagic/livyclientlib/exceptions.py#L41